docs/articles/semantic_chunking.md

```diff
@@ -0,0 +1,517 @@
+<!-- SEO: Explore semantic chunking for Retrieval Augmented Generation (RAG) in this comprehensive guide. Learn about embedding similarity, hierarchical clustering, and LLM-based methods for optimal text chunking. Discover how semantic chunking improves RAG performance compared to traditional rule-based approaches. Includes code examples, evaluation metrics, and comparisons using HotpotQA and SQUAD datasets with BAAI/bge-small-en-v1.5 embeddings.
+-->
+
+# Semantic Chunking
+
+Chunking is simply dividing data into smaller manageable pieces. Similar to how it is easier for a human to cognitively manage smaller pieces of information, it is easier for a computer to work with information divided into smaller chunks. In the context of Retrieval Augmented Generation (RAG), chunking plays a great role, especially when it comes to large datasets. The large dataset is divided into chunks and stored in vector databases and indexes. When there is a user query, it is easier to skim through and retrieve the most relevant chunks only without hassle.
+
+There are various ways we can chunk a document. The choice of chunking method can significantly impact the performance of a RAG system, depending on the nature of the data and the specific RAG implementation. Some widely used chunking algorithms include fixed character splitter, recursive character splitter, document-specific splitter, semantic splitter, and agentic splitter.
+
+In this article, we will explore the semantic chunking algorithm in detail and comprehensively evaluate it using different embedding models and datasets.
+
+
+## What is Semantic Chunking? 
+
+The idea of semantic chunking is pretty simple. The semantic splitter adaptively picks the breakpoint in between sentences by comparing embedding similarity. This ensures that each chunk contains semantically cohesive sentences. 
+
+Although there can be variations, the algorithm typically uses a sliding window approach to compare adjacent sentences. It calculates the cosine similarity between the embeddings of consecutive sentences, establishing a threshold to determine chunk boundaries. When the similarity drops below this threshold, it signals a shift in semantic content and marks a breakpoint. 
+
+The workflow basically consists of 3 steps:
+
+![Full text](../assets/use_cases/semantic_chunking/sample1.png)
+
+1.	Split the text into sentences.
+
+    ![Text split](../assets/use_cases/semantic_chunking/sample2.png)
+
+2.	Generate embeddings for the sentences.
+
+    ![Embedding split](../assets/use_cases/semantic_chunking/sample3.png)
+
+3. Group similar sentences comparing their embeddings.
+
+    ![Chunks](../assets/use_cases/semantic_chunking/sample4.png)
+    
+## Types of Semantic Chunking
+
+The document's semantics can be inferred in various ways for optimal chunking. Here are 3 ways you can implement semantic chunkers.
+

```

Comment:
can you provide a rationale for choosing these three (below)?

Reply:
You can add the sentence, 

"The upcoming sections will present evaluation results, helping you select the most appropriate approach for your application."

Reply:
@Ashish-Abraham there are many semantic chunking methods - many more beyond these three. You've evaluated three below. Why these three instead of the others? That's the rationale that's missing here.

Reply:
Yes. Better to put it like we're evaluating three popular ones